client/cli/proc-macro/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use proc_macro_crate::crate_name;
+use quote::quote;
+use syn::{Error, Expr, Ident, ItemFn};
+
+/// Macro that inserts a span with the node name at the beginning of the function. This prefix all
```

Comment:
You should also mention that it only applies to stuff that is run inside the function and how it propagates (TaskManager etc)

Reply:
done